framework/src/tinyalsa/tinyalsa.c

```diff
@@ -311,6 +308,16 @@ int pcm_set_config(struct pcm *pcm, const struct pcm_config *config)
 		pcm->config = *config;
 	}
 
+	if (ioctl(pcm->fd, AUDIOIOC_GETBUFFERINFO, (unsigned long)&buf_info) < 0) {
+		/* Driver doesn't report it's buffer size.  Use our default. */
+		buf_info.buffer_size = CONFIG_AUDIO_BUFFER_NUMBYTES;
+		buf_info.nbuffers = CONFIG_AUDIO_NUM_BUFFERS;
+	}
+
+	pcm->config.period_size = buf_info.buffer_size;
+	pcm->config.period_count = buf_info.nbuffers;
+	pcm->buffer_size = pcm->config.period_size;
+
```

Comment:
Since we are always overwriting the period_size and period_count values, it means we wont use values passed by user for period size and count. So, its better we move these variables to pcm struct.

Reply:
Currently it is not problem, I want to keep same as original tinyasa